### PR TITLE
Add individual reward claiming

### PR DESCRIPTION
**Pull request by sisby-folk**
Addresses a small point in #68 

Adds the loot bag button to all reward types, instead of just selectable. Adds a new packet to handle claiming a single reward, and a method in quest to help out.

Integrated into BaseItemReward, which selectable now also uses.

https://github.com/terrarium-earth/Heracles/assets/55819817/73dd31e0-2b6f-4aa2-a4e0-1e42ae811c64

### CODE REVIEW

**Comment by sisby-folk** [comment]
Here's another shot showing the individual claiming better

https://github.com/terrarium-earth/Heracles/assets/55819817/f93bcfd0-a5c3-425e-be97-9f9b857c21a2

The screen closing here isn't actually required, I'm just showing the chat.

File: common/src/main/java/earth/terrarium/heracles/Heracles.java [hunk 1]
```diff
@@ -3,12 +3,15 @@
...
 import net.minecraft.core.RegistryAccess;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.nio.file.Path;
 import java.util.function.Supplier;
 
 public class Heracles {
     public static final String MOD_ID = "heracles";
+    public static final Logger LOGGER = LoggerFactory.getLogger(MOD_ID);
```
Comment (ThatGravyBoat) [inline common/src/main/java/earth/terrarium/heracles/Heracles.java]: Please use the Mojang method `LogUtils.getLogger()` instead

File: common/src/main/java/earth/terrarium/heracles/common/network/packets/rewards/ClaimRewardPacket.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import com.teamresourceful.resourcefullib.common.networking.base.PacketContext;
+import com.teamresourceful.resourcefullib.common.networking.base.PacketHandler;
+import earth.terrarium.heracles.Heracles;
+import earth.terrarium.heracles.api.quests.Quest;
+import earth.terrarium.heracles.common.handlers.quests.QuestHandler;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerPlayer;
+
+public record ClaimRewardPacket(String quest, String reward) implements Packet<ClaimRewardPacket> {
```
Comment (ThatGravyBoat) [inline common/src/main/java/earth/terrarium/heracles/common/network/packets/rewards/ClaimRewardPacket.java]: I would perfer if this is combined with the ClaimRewardsPacket and have the reward string be optional and if its present just claim that reward if its not claim all

**Review by ThatGravyBoat — CHANGES_REQUESTED** [review]

File: common/src/main/java/earth/terrarium/heracles/common/network/packets/rewards/ClaimRewardPacket.java (hunk 2 above)
Comment (sleepingdragoninn) [reply common/src/main/java/earth/terrarium/heracles/common/network/packets/rewards/ClaimRewardPacket.java]: done up using a blank string in 2a725e9 - I didn't wanna mix nulls and empty strings in the buffer, but if you have a better way to do it let me know.

File: common/src/main/java/earth/terrarium/heracles/Heracles.java (hunk 1 above)
Comment (sleepingdragoninn) [reply common/src/main/java/earth/terrarium/heracles/Heracles.java]: done in 2a725e9

File: common/src/main/java/earth/terrarium/heracles/common/network/packets/rewards/ClaimRewardPacket.java (hunk 2 above)
Comment (ThatGravyBoat) [reply common/src/main/java/earth/terrarium/heracles/common/network/packets/rewards/ClaimRewardPacket.java]: An empty string works perfectly fine

**Review by ThatGravyBoat — APPROVED** [review]

**Pull request merged**